run/helm/snap/templates/daemonset.yaml

```diff
@@ -0,0 +1,43 @@
+apiVersion: extensions/v1beta1
+kind: DaemonSet
+metadata:
+  name: snap
+  namespace: {{ .Release.Namespace }}
+  labels:
+    chart: "{{ .Chart.Name }}-{{ .Chart.Version }}"
+spec:
+  template:
+    metadata:
+      name: snap
+      labels:
+        daemon: snapteld
+    spec:
+      containers:
+      - name: {{ .Chart.Name }}
+        image: "{{ .Values.image.repo }}:{{ .Values.image.tag }}"
+        imagePullPolicy: {{ .Values.image.pullPolicy }}
+        securityContext:
+          privileged: {{ .Values.privileged }}
+        ports:
+        - containerPort: 8181
+        volumeMounts:
+        - name: autoload
+          mountPath: {{ .Values.config.autodiscover_path }}
+        - name: config
+          mountPath: /etc/snap
+        {{- range $key, $val := .Values.mounts }}
+        - name: {{ $key | replace "/" "" | replace "." "" }}
+          mountPath: {{ $val | quote }}
+        {{- end }}
+      volumes:
+      - name: autoload
+        hostPath:
```

Comment:
@pskrzyns I am not completely understanding the argument of "we will need to support all possible storage solutions." Could you clarify what you mean for me? 

From my experience with this, PVCs are just abstractions to attach to an exposed persistent volumes and storage classes. Many Helm charts have PVCs defined with a variable to use or not use persistent storage.

Beyond the PVC debate, I agree with @jose5918, we need a better solution for managing the plugins and tasks that are loaded into Snap, which is an unsolved problem as of yet. It might be a good thing to discuss what those options are.